CondCore/SiPixelPlugins/plugins/SiPixelDynamicInefficiency_PayloadInspector.cc

```diff
@@ -104,6 +113,112 @@ namespace {
       }
       return false;
     }
+
+    //_________________________________________________
+    double getMatchingGeomFactor(const DetId& detid,
+                                 const std::map<unsigned int, double>& map_geomfactor,
+                                 const std::vector<uint32_t>& detIdmasks) {
+      double geomfactor_db = 1;
+      for (auto map_element : map_geomfactor) {
+        const DetId mapid = DetId(map_element.first);
+        if (mapid.subdetId() != detid.subdetId())
+          continue;
+        size_t __i = 0;
+        for (; __i < detIdmasks.size(); __i++) {
+          DetId maskid = DetId(detIdmasks.at(__i));
+          if (maskid.subdetId() != mapid.subdetId())
+            continue;
+          if ((detid.rawId() & maskid.rawId()) != (mapid.rawId() & maskid.rawId()) &&
+              (mapid.rawId() & maskid.rawId()) != DetId(mapid.det(), mapid.subdetId()).rawId())
+            break;
+        }
+        if (__i != detIdmasks.size())
+          continue;
+        geomfactor_db *= map_element.second;
+      }
+      return geomfactor_db;
+    }
+
+    //_________________________________________________
+    std::vector<double> getMatchingPUFactors(const DetId& detid,
+                                             const std::map<unsigned int, std::vector<double> >& map_pufactory,
+                                             const std::vector<uint32_t>& detIdmasks) {
+      std::vector<double> pufactors_db;
+      for (const auto& map_element : map_pufactory) {
+        const DetId mapid = DetId(map_element.first);
+        if (mapid.subdetId() != detid.subdetId())
+          continue;
+        size_t __i = 0;
+        for (; __i < detIdmasks.size(); __i++) {
+          DetId maskid = DetId(detIdmasks.at(__i));
+          if (maskid.subdetId() != mapid.subdetId())
+            continue;
+          if ((detid.rawId() & maskid.rawId()) != (mapid.rawId() & maskid.rawId()) &&
+              (mapid.rawId() & maskid.rawId()) != DetId(mapid.det(), mapid.subdetId()).rawId())
+            break;
+        }
+        if (__i != detIdmasks.size())
+          continue;
+        pufactors_db = map_element.second;
+      }
+      return pufactors_db;
+    }
+
+    /* 
+    //(Not used for the moment)
```

Comment:
Even if currently not used, I wonder what is the purpose of commenting out the definition of this method: was it giving problems while building?

Reply:
@perrotta 

indeed, it leads to the following compilation warning:

```bash
/tmp/musich/CMSSW_13_0_X_2023-01-10-2300/src/CondCore/SiPixelPlugins/plugins/SiPixelDynamicInefficiency_PayloadInspector.cc:169:10: warning: 'bool {anonymous}::SiPixDynIneff::matches(const DetId&, const DetId&, const std::vector<unsigned int>&)' defined but not used [-Wunused-function]
  169 |     bool matches(const DetId& detid, const DetId& db_id, const std::vector<uint32_t>& DetIdmasks) {
      |          ^~~~~~~
```
which is a bit annoying.
OTOH with these changes

```diff
diff --git a/CondCore/SiPixelPlugins/plugins/SiPixelDynamicInefficiency_PayloadInspector.cc b/CondCore/SiPixelPlugins/plugins/SiPixelDynamicInefficiency_PayloadInspector.cc
index 043b6459248..9c8d1701cd3 100644
--- a/CondCore/SiPixelPlugins/plugins/SiPixelDynamicInefficiency_PayloadInspector.cc
+++ b/CondCore/SiPixelPlugins/plugins/SiPixelDynamicInefficiency_PayloadInspector.cc
@@ -164,10 +164,9 @@ namespace {
       return pufactors_db;
     }
 
-    /* 
     //(Not used for the moment)
     //_________________________________________________
-    bool matches(const DetId& detid, const DetId& db_id, const std::vector<uint32_t>& DetIdmasks) {
+    [[maybe_unused]] bool matches(const DetId& detid, const DetId& db_id, const std::vector<uint32_t>& DetIdmasks) {
       if (detid.subdetId() != db_id.subdetId())
         return false;
       for (size_t i = 0; i < DetIdmasks.size(); ++i) {
@@ -180,7 +179,6 @@ namespace {
       }
       return true;
     }
-    */
 
     //_________________________________________________
     bool checkPhase(const SiPixelPI::phase phase, const std::vector<uint32_t>& masks_db) {

```

I see that the warning goes away.
I can apply this in the next iteration (if it will still be unused).